scripts/postbuild.ts

```diff
@@ -38,3 +38,13 @@ Object.keys(pkgJson.exports)
   });
 
 writeFileSync(dtsPath, dtsContents);
+
+/**
+ * 3). Run typecheck
+ */
+try {
+  execSync('tsc --noEmit', { stdio: 'inherit' });
```

Comment:
Why do we need this? Doesn't step `1) Emit Declaration Maps` already do type checking?

Reply:
> Doesn't step 1) Emit Declaration Maps already do type checking?

Not from my experiences - I can't find the documentation on why this is. You can try for yourself and add a type error within a `test` file and comment out this step, it should pass compilation.

Reply:
It's because all `tsconfig.dts.ts` files exclude test files by default.

It feels off that we're doing type checking _again_ here just so that we catch compilation errors in our tests. I think that having a `type:check-tests` command run only for `**.test.ts` files in `pretest` for every package would be both faster and separate concerns better:
```json
// tsconfig.test.json
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "noEmit": true
  },
  "include": ["**/*.test.ts"]
}

// package.json
{
  ...
   "scripts": {
    "pretest": "run-s some-commands type:check",
    "type:check-tests": "tsc -p tsconfig.test.json",
  },
}
```

Reply:
You're right, and this feel more correct - I will make these changes.

Reply:
I moved this to the `lint` hook
961e0787007f952b871caa757c9ff255e813ff4e